tests/util.py

```diff
@@ -146,3 +146,10 @@ def check_rindex_entry(entry, name, offset=None, flags=None):
 
     if flags is not None:
         assert e_flags == flags
+
+
+def generate_path(dirname, filename, encoding=None):
+    path = os.path.join(str(dirname), filename)
+    if encoding is not None:
```

Comment:
This is the case where:

    if encoding:
        use it...

Is better, since empty string is not a valid encoding.